libs/base/Control/Monad/Reader/Reader.idr

```diff
@@ -42,35 +42,33 @@ runReader s = runIdentity . runReaderT s
 
 public export
 implementation Functor f => Functor (ReaderT stateType f) where
-  map f (MkReaderT g) = MkReaderT (\st => map f (g st))
+  map f (MkReaderT g) = MkReaderT \st => f <$> g st
```

Comment:
Trailing binders (and so lambdas too) are going to be removed (cf. 1711) so that
all the arguments are treated uniformly & we can generate better error messages
in type signatures.

Reply:
I've rebased upon just merged 1711.